operator/redisfailover/handler.go

```diff
@@ -77,6 +78,19 @@ func (r *RedisFailoverHandler) Handle(_ context.Context, obj runtime.Object) err
 			r.mClient.SetClusterError(rf.Namespace, rf.Name)
 			return err
 		}
+	} else {
+		// If the versions do not match, this signifies that the resource has been updated and therefore, the status also requires updating.
+		if rf.GetObjectMeta().GetGeneration() != rf.Status.ObservedGeneration {
+
+			rf.Status.Conditions = []redisfailoverv1.ClusterCondition{}
+			rf.Status.AddCondition(redisfailoverv1.ClusterCondition{
+				Status:  redisfailoverv1.ConditionTrue,
+				Type:    redisfailoverv1.AppStatePending,
+				Message: "RedisFailover reconciling...",
+			})
+			rf.Status.ObservedGeneration = rf.GetObjectMeta().GetGeneration()
+			r.rfService.UpdateStatus(rf)
```

Comment:
Should we check for error here like we do other places?

rf, err := r.rfService.UpdateStatus(rf)